src/WebJobs.Extensions.OpenAI/Assistants/AssistantService.cs

```diff
@@ -184,14 +188,11 @@ public async Task<AssistantState> PostMessageAsync(AssistantPostAttribute attrib
             throw new ArgumentException("The assistant must have a user message", nameof(attribute));
         }
 
-        if (this.tableClient is null)
-        {
-            throw new ArgumentException("The assistant must be initialized first using CreateAssistantAsync", nameof(this.tableClient));
-        }
-
         this.logger.LogInformation("Posting message to assistant entity '{Id}'", attribute.Id);
 
-        InternalChatState? chatState = await this.LoadChatStateAsync(attribute.Id, cancellationToken);
+        this.tableClient = this.GetOrCreateTableClient(attribute.ChatStorageConnectionSetting, attribute.CollectionName);
```

Comment:
This should be assigning a local variable instead of assigning `this.tableClient`. The `this.tableClient` member variable should only be assigned and used by `this.GetOrCreateTableClient(...)`.